### ISSUE TITLE
Restarting database container clobbers database

### ISSUE THREAD

**Issue by BenjaminPelletier** [RopeWiki/app#29]
Currently, even when the database container is merely stopped then started, the database is found to be broken after restarting:

```
docker container stop dev_ropewiki_db_1
docker container start dev_ropewiki_db_1
```

Database content should persist in the named volume through even container removals.

**Comment by moaz178** [comment]
You can make use of Docker volumes. By using a named volume, you can separate the data from the container, allowing it to be preserved even when the container is stopped, started, or removed.

docker volume create ropewiki_db_data

docker run -d --name dev_ropewiki_db_1 -v ropewiki_db_data:/var/lib/postgresql/data <your-database-image>

When you stop and start the container or remove and recreate it, the data will persist in the named volume. The database container will use the same volume for data storage, ensuring that the content is preserved across container lifecycles.

**Comment by hcooper** [comment]
Hi @moaz178 - the db is already using docker volumes: https://github.com/RopeWiki/app/blob/master/docker-compose.yaml#L26-L27

Ben dug out some more logs, and the problem is probably better described as "webserver can't connect to database after database is restarted".

```
{"log":"2023-05-23T07:24:49.335605Z 4 [Note] Access denied for user 'ropewiki'@'172.18.0.3' (using password: YES)\n","stream":"stderr","time":"2023-05-23T07:24:49.336477954Z"}
```

My theory is that it's related to the workflow around `deploy_tool.py`. It first creates an initial database, and starts the webserver using placeholder credentials, then restores a production backup over the top. I think the problem arises because the db doesn't have privileges flushed until it's restarted.

A way to confirm or deny this theory is to restore the database, testing the website is working, manually flush privileges, then test again. If my theory is correct, the site will be broken after the flush.

**Comment by hcooper** [comment]
Think this has been finally resolved, as detailed here: https://github.com/RopeWiki/app/wiki/Resolving-the-repeated-post-migration-database-problem

tl;dr - corrupt `mysql.user` table schemas (probably from migrating between different mysql versions).
